base/decfp.jl

```diff
@@ -0,0 +1,32 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# These define the types for IEEE 754-2008 Decimal Floating Point formats
+# Support for these types is available via packages, such as DecFP.jl
+
+export AbstractFlt, BinaryFloat, DecimalFloat, DecimalFloatBinary, DecimalFloatPacked
+
+"""Abstract types for all floating point types"""
+abstract AbstractFlt{b} <: Real
+typealias BinaryFloat AbstractFlt{2}
+typealias DecimalFloat AbstractFlt{10}
+abstract DecimalFloatBinary <: DecimalFloat
+abstract DecimalFloatPacked <: DecimalFloat
+
+export isIEEE
+isIEEE{T <: Number}(::Type{T}) = false
+for w in (32, 64, 128)
+    BID = symbol(:d, w)
+    DPD = symbol(:D, w)
+    @eval begin
+        export $BID, $DPD
+        #"""IEEE 754-2008 $w-bit Decimal Floating Point, binary format"""
+        bitstype $w $BID <: DecimalFloatBinary
+        #"""IEEE 754-2008 $w-bit Decimal Floating Point, packed format"""
+        bitstype $w $DPD <: DecimalFloatPacked
+        isIEEE(::Type{$BID}) = true
+        isIEEE(::Type{$DPD}) = true
+    end
+end
+for w in (16, 32, 64)
+    @eval isIEEE(::Type{$(symbol(:Float,w))}) = true
```

Comment:
More fundamentally, I wouldn't add this `isIEEE()` function until we really find a use for it. I fail to see what difference it could make for programming whether these types follow the IEEE standards or not. They are so different.